dataset/bin.cpp

```diff
@@ -453,9 +453,32 @@ DataArray groupby_concat_bins(const DataArrayConstView &array,
       builder.groups(), {reductionDim});
 }
 
+namespace {
+void validate_bin_args(const std::vector<VariableConstView> &edges,
+                       const std::vector<VariableConstView> &groups) {
+  if (edges.empty()) {
+    if (groups.empty()) {
+      throw except::BucketError(
+          "Arguments 'edges' and 'groups' of scipp.bin are "
+          "both empty. At least one must be set.");
+    }
+  } else {
+    for (std::size_t i = 0; i < edges.size(); ++i) {
+      if (const auto &shape = edges[i].dims().shape();
+          std::any_of(shape.begin(), shape.end(),
+                      [](const scipp::index s) { return s == 1; })) {
+        throw except::BucketError("Not enough bin edges in dim " +
+                                  std::to_string(i) + ". Need at least 2.");
+      }
+    }
+  }
+}
+} // namespace
+
 DataArray bin(const DataArrayConstView &array,
               const std::vector<VariableConstView> &edges,
               const std::vector<VariableConstView> &groups) {
+  validate_bin_args(edges, groups);
```

Comment:
I think there are more places below where `validate_bin_args` should be called? e.g. in the next `bin` overload?

Reply:
That would validate twice, the `bin` lower down is just called by this.

Reply:
That might be true at the moment. But the function is still exposed in the header. So not validating would not be completely safe.

Reply:
Does anything break if we *only* validate in the templated `bin`?

Reply:
This one
https://github.com/scipp/scipp/blob/96f454389fa25ab0271a490961c88409e8d79fd3/dataset/bin.cpp#L516
or this one?
https://github.com/scipp/scipp/blob/96f454389fa25ab0271a490961c88409e8d79fd3/dataset/bin.cpp#L111

Reply:
The one from line 516.